homeassistant/components/onvif/parsers.py

```diff
@@ -12,16 +14,13 @@
 PARSERS: Registry[str, Callable[[str, Any], Coroutine[Any, Any, Event]]] = Registry()
 
 
-def datetime_or_zero(value: str) -> datetime:
-    """Convert strings to datetimes, if invalid, return datetime.min."""
+def datetime_or_zero(value: str) -> datetime | None:
```

Comment:
```suggestion
def datetime_or_zero(value: str) -> datetime.datetime | None:
```

Reply:
or import `datetime as dt` and then use `dt.datetime` to avoid confusion

Reply:
And should we exit earlier every time we call `dt_util.as_local(date_time)` as that doesn't like None datetimes

Reply:
Side note: it seems that onvif is part of the `mypy` ignore list so we're missing out on some checks

Reply:
> exit earlier every time we call `dt_util.as_local(date_time)` as that doesn't like nullable datetimes

Good point, let's fix that too 👍